crates/proof/executor/src/builder/env.rs

```diff
@@ -80,31 +100,35 @@ where
+    ///
+    /// ## Note
+    /// Before Jovian activation, the min-base-fee is 0.
     pub(crate) fn active_base_fee_params(
         config: &RollupConfig,
         parent_header: &Header,
         payload_attrs: &OpPayloadAttributes,
-    ) -> ExecutorResult<BaseFeeParams> {
-        let base_fee_params =
-            if config.is_holocene_active(payload_attrs.payload_attributes.timestamp) {
-                // After Holocene activation, the base fee parameters are stored in the
-                // `extraData` field of the parent header. If Holocene wasn't active in the
-                // parent block, the default base fee parameters are used.
-                config
-                    .is_holocene_active(parent_header.timestamp)
-                    .then(|| decode_holocene_eip_1559_params(parent_header))
-                    .transpose()?
-                    .unwrap_or(config.chain_op_config.as_canyon_base_fee_params())
-            } else if config.is_canyon_active(payload_attrs.payload_attributes.timestamp) {
-                // If the payload attribute timestamp is past canyon activation,
-                // use the canyon base fee params from the rollup config.
-                config.chain_op_config.as_canyon_base_fee_params()
+    ) -> ExecutorResult<(BaseFeeParams, u64)> {
+        if config.is_holocene_active(payload_attrs.payload_attributes.timestamp) {
+            // After Holocene activation, the base fee parameters are stored in the
+            // `extraData` field of the parent header. If Holocene wasn't active in the
+            // parent block, the default base fee parameters are used.
+            if config.is_jovian_active(parent_header.timestamp) {
+                decode_jovian_eip_1559_params_block_header(parent_header)
+            } else if config.is_holocene_active(parent_header.timestamp) {
+                decode_holocene_eip_1559_params_block_header(parent_header)
+                    .map(|base_fee_params| (base_fee_params, 0))
             } else {
-                // If the payload attribute timestamp is prior to canyon activation,
-                // use the default base fee params from the rollup config.
-                config.chain_op_config.as_base_fee_params()
-            };
-
-        Ok(base_fee_params)
+                Ok((config.chain_op_config.as_canyon_base_fee_params(), 0))
+            }
+        } else if config.is_canyon_active(payload_attrs.payload_attributes.timestamp) {
+            // If the payload attribute timestamp is past canyon activation,
+            // use the canyon base fee params from the rollup config.
+            Ok((config.chain_op_config.as_canyon_base_fee_params(), 0))
+        } else {
+            // If the payload attribute timestamp is prior to canyon activation,
+            // use the default base fee params from the rollup config.
+            Ok((config.chain_op_config.as_base_fee_params(), 0))
+        }
```

Comment:
this block of code seems weird, how about just checking jovian up top, etc. and returning directly and skipping the if elses